src/core/lib/event_engine/posix_engine/posix_engine.h

```diff
@@ -264,17 +242,36 @@ class PosixEventEngine final : public PosixEventEngineWithFdSupport {
 #if defined(GRPC_POSIX_SOCKET_TCP) && \
     !defined(GRPC_DO_NOT_INSTANTIATE_POSIX_POLLER)
 
+  // A helper class to manage lifetime of the poller associated with the
```

Comment:
This doesn't manage the lifetime of a poller. It's also a subset of the `ThreadPool` interface itself, the only difference being a Quiesce method. Can we just use a `shared_ptr<ThreadPool>` instead?